src/device.c

```diff
@@ -467,15 +474,12 @@ int device_clear_credentials_on_file(void)
 int device_start_event(void)
 {
 	int rc;
-
-	rc = event_start(on_event_timeout);
+	rc = event_start(on_event_timeout, thing.time_sec);
 	if (rc < 0) {
 		l_error("Failed to start event");
 		return rc;
 	}
 
-	l_hashmap_foreach(thing.data_items, foreach_event_add_data_item, NULL);
-
 	return 0;
```

Comment:
What does the zero means? I think the number zero should be replaced by a variable with a explicit name